test/e2e/page-objects/pages/confirmations/redesign/permit-confirmation.ts

```diff
@@ -0,0 +1,142 @@
+import { strict as assert } from 'assert';
+import { Driver } from '../../../../webdriver/driver';
+import { DAPP_HOST_ADDRESS } from '../../../../constants';
+import Confirmation from './confirmation';
+
+export default class PermitConfirmation extends Confirmation {
+  constructor(driver: Driver) {
+    super(driver);
+
+    this.driver = driver;
+  }
+
+  private originSelector = { text: DAPP_HOST_ADDRESS };
+
+  private contractPetNameSelector = {
+    css: '.name__value',
+    text: '0xCcCCc...ccccC',
+  };
+
+  private primaryTypeSelector = { text: 'Permit' };
+
+  private ownerSelector = { css: '.name__name', text: 'Account 1' };
+
+  private spenderSelector = { css: '.name__value', text: '0x5B38D...eddC4' };
+
+  private valueSelector = { text: '3,000' };
+
+  private nonceSelector = { text: '0' };
+
+  private deadlineSelector = { text: '09 June 3554, 16:53' };
+
+  private nftContractPetNameSelector = {
+    css: '.name__value',
+    text: '0x581c3...45947',
+  };
+
+  private nftTitle = { text: 'Withdrawal request' };
```

Comment:
[suggestion/non-blocker]: rather than private vars here, we could keep them outside of the class as a map structure. This would provide a quick lookup and instantiating a single object

e.g.
```
const SELECTOR = {
  nft: { text: 'Withdrawal request' },
  nftContractPetName: {
    css: '.name__value',
    text: '0x581c3...45947',
  }
}

export default class PermitConfirmation extends Confirmation {
  // ..

  .. await this.driver.findElement(SELECTOR.nft);
```

Reply:
That's a good idea. But maybe best to do it out of the scope of this PR as there are already many pages using the current pattern for now. I'd also like to run it by the extension team.